modules/management/src/main/java/org/mule/module/management/agent/DefaultJmxSupportAgent.java

```diff
@@ -116,13 +117,13 @@ public void initialise() throws InitialisationException
             // there must be only one jmx agent, so lookup by type instead
             if (registry.lookupObject(AbstractJmxAgent.class) == null)
             {
-                registry.registerAgent(agent);
```

Comment:
because when you manually register something ,spring applies the completed phases, but at this point no phase has yet been completed. It's the same as when a flow initialises its message source